Add the ability to report cluster-metrics only from the master-node

alfasin opened the pull request:
After our discussion I decided to split the implementation into two PRs:

1. This one: which adds the ability to report cluster-related metrics from the master-node only, in order to avoid multiple reports of the same metric. 
This PR leaves the current behavior as default, and in order to activate this new-behavior, the configuration should have the property: 
Raigad.report.metrics.from.master.only=true

2. The next PR that I'll submit will have the additional logic for Node-Healthcheck (reports the metric: "is-elasticsearch-process-running" and has an additional endpoint-url that responds to REST calls with this value as well).

sagarl commented on raigad/src/main/java/com/netflix/raigad/monitoring/NodeHealthMonitor.java:
```diff
@@ -0,0 +1,105 @@
+            healthBean.esprocessdown = 0;
+            if (!ElasticsearchProcessMonitor.isElasticsearchRunning()) {
+                healthBean.esprocessdown = 1;
+            }
```
Add healthBean.esprocessdown = 0; to else statement. So that it gets executed only when ES process is Up. 

No need to set the value twice if ES is not Running.

sagarl commented on raigad/src/main/java/com/netflix/raigad/monitoring/NodeHealthMonitor.java:
```diff
@@ -0,0 +1,105 @@
+            healthBean = new AtomicReference<HealthBean>(new HealthBean());
+        }
+
+        @Monitor(name = "es_isesprocessdown", type = DataSourceType.GAUGE)
```
Name of variable is bit misnomer. "es_isesprocessdown"

Value of es_isesprocessdown is set to 0, when ES is Up.

sagarl commented on raigad/src/main/java/com/netflix/raigad/monitoring/HealthMonitor.java:
```diff
@@ -61,13 +65,19 @@ public HealthMonitor(IConfiguration config,InstanceManager instanceManager)
+        // In case we configured only the Master-node to report metrics
+        // and this node is not a master - bail out
+        if (config.reportMetricsFromMasterOnly() && EsUtils.amIMasterNode(config, httpModule)) {
+            return;
```
This check will work for all nodes except Master. 
I think, it should be as follows
if (config.reportMetricsFromMasterOnly() && !EsUtils.amIMasterNode(config, httpModule))

alfasin commented:
Good catch for the missing "not" condition!
I changed renamed the metric and added the logic to run HealthMonitor only in one of the following cases:

- non ASG deployment
- ASG deployment + this node is a master
- ASG deployment + configuration setting (run on master only) is set to false (default behavior)
